src/lib/tools.ts

```diff
@@ -0,0 +1,313 @@
+/*
+Copyright 2019 Balena Ltd.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+import * as Bluebird from 'bluebird';
+import * as _ from 'lodash';
+import * as fs from 'mz/fs';
+import { dirname, isAbsolute, join, resolve } from 'path';
+import * as tunnel from 'tunnel-ssh';
+import {
+	FileLoadError,
+	NotImplementedError,
+	UnsupportedError,
+	URILoadError,
+} from './error-types';
+
+import { TimeoutError } from 'bluebird';
+import * as yamljs from 'yamljs';
+import { ConfigMap } from './controllers/config-store/config-store';
+
+export const tunnelAsync = Bluebird.promisify(tunnel);
+
+/**
+ * Returns an absolute path for a path, when in basePath
+ * @param {string} path: The file path, (or an absolute path)
+ * @param {string} basePath: The base path
+ * @returns {string} An absolute path
+ */
+export function getAbsolutePath(path: string, basePath: string): string {
+	return isAbsolute(path) ? path : join(basePath, path);
+}
+
+/**
+ * Gets absolute URI
+ * @param {string} uri
+ * @param {string} basePath
+ * @returns {string}
+ */
+export function getAbsoluteUri(uri: string, basePath: string): string {
+	if (isValidGitUri(uri)) {
+		throw new NotImplementedError('Git URI support not implemented yet');
+	} else if (isLocalPathUri(uri)) {
+		return getAbsolutePath(uri, basePath);
+	} else {
+		throw new UnsupportedError('URI type not supported yet');
+	}
+}
+
+/**
+ * Loads a yaml file as object
+ * @param {string} filePath
+ * @param {string} errorMessage
+ * @returns {Promise<string>}
+ */
+export async function loadFromFile(
+	filePath: string,
+	errorMessage: string = '',
+): Promise<object> {
+	try {
+		return yamljs.parse(await fs.readFile(filePath, 'utf8'));
```

Comment:
Switching to [js-yaml](https://www.npmjs.com/package/js-yaml) is probably better as it's what we use elsewhere and is more widely used/maintained, and also provides a [safeLoad](https://www.npmjs.com/package/js-yaml#safeload-string---options-) function which we'd probably want to use here

Reply:
Agreed and done across the codebase 👍